src/rgw/rgw_data_sync.cc

```diff
@@ -5168,6 +5178,8 @@ class RGWSyncBucketCR : public RGWCoroutine {
   rgw_bucket_index_marker_info info;
 
   RGWSyncTraceNodeRef tn;
+  rgw_bucket_index_marker_info remote_info;
```

Comment:
we already have `rgw_bucket_index_marker_info info;` initialized above, so we don't need to fetch it again here. this `info` gets passed into `InitBucketFullSyncStatusCR`, so we can use `markers.from_string(info.max_marker, -1)` internally to get the shard count associated with this `info.latest_gen` generation

`InitBucketFullSyncStatusCR` has two different places that need `num_shards`. in the `check_compat` case, the call to `CheckAllBucketShardStatusIsIncremental` actually wants `num_shards` from generation 0 - we can look that up from the local bucket's log layouts

in my 'bucket sync status' PR, i do something similar for backward compat: https://github.com/ceph/ceph/pull/42872/files#diff-f3b81fb31115c2616085efcfd6a8daed67a94d1041a03af5de48cc572978f547R2327-R2339

Reply:
I'm confused by:

> we already have `rgw_bucket_index_marker_info info;` initialized above, so we don't need to fetch it again here.

It looks like the only initialization is on line 5372? I don't see this being initialized in the constructor or anywhere before that.

Reply:
it's initialized above in the call to `RGWReadRemoteBucketIndexLogInfoCR`: https://github.com/ceph/ceph/blob/wip-rgw-multisite-reshard/src/rgw/rgw_data_sync.cc#L5256-L5263

Reply:
Oh, I see what you mean. I thought you were pointing out that that variable was initialized already, not that we had an entirely superfluous variable.